Plasmamen get plasma in their box

Michael-Ashfield opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Plasmamen currently get emergency oxygen in their survival boxes, this is completely useless to the Plasmaman, this replaces it with a belt tank of plasma which they can actually use.
Survival boxes now check if the player spawns as a plasmaman and gives them plasma instead of oxygen if they do.

## Why It's Good For The Game

Plasmamen getting oxygen was useless, now they can actually use their survival box.

## Changelog
:cl: Mickyy5
add: Nanotrasen are now issuing Plasmamen with plasma in their survival boxes
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Commit:
Updated this old code to fork

Commit:
rebase branch

Commit:
Revert "rebase branch"

This reverts commit 72e70fe8ccabfa6c750193a57beeec14b30a2533.

Commit:
gives the clown hugbox

Commit:
Clown gets hugbox back

Shadowflame909 commented:
epic

loserXD commented:
finally

coiax commented:
This seems... inelegant in its execution.

Why not just put a single "isplasmaman" check in the equip code, replacing oxygen tanks with plasma tanks, instead of making a whole bunch of paths?

Michael-Ashfield commented:
I was informed on discord that changing contents within stuff is janky, additionally theres a few weird things with engineers having their special tanks and not all survival box paths looking like /obj/item/storage/box/survival. 
I'm new to this but if you think that's a better way I will look into it and see if I can figure it out.

Michael-Ashfield commented:
Okay I think I thought of a way to do it like that, I will check on discord if Im just stupid first

Commit:
Removed new boxes, changed box code

Michael-Ashfield commented:
**Update-**
Removed all the new boxes and added checks into the old boxes that checks if the user is a plasma man, tested. thanks to Space Prius for the help.
Now only affects 1 file

Commit:
Sorted line spaces

coiax commented on code/game/objects/items/storage/boxes.dm:
```diff
@@ -124,25 +124,41 @@
 	new /obj/item/reagent_containers/hypospray/medipen(src)
+	
+	var/mob/living/M = src.loc
+	if(!isplasmaman(M))
```
This will runtime if the box is ever instanced not inside a mob, probably resulting in boxes not in mobs missing oxygen tanks.

nemvar replied:
Since when do istype checks runtime if the argument is null and/or the wrong type?

PKPenguin321 replied:
I don't think it will runtime, but it definitely doesn't need to be typecasted

Michael-Ashfield replied:
actual real Maintainer on Discord advised I just use `if(!isplasmaman(loc))`, I tested this and it seemed to work both in my outfit and when spawning survival boxes with the admin debug tools. No runtimes on the debug runtime tool detected.

Commit:
Removed sauce

ExcessiveUseOfCobblestone commented:
haven't done my homework but how is the loc a mob? It should be spawned as part of the backpack which would be on the mob, it shouldn't be going into the mob then in the backpack.

Michael-Ashfield commented:
actual real Maintainer on Discord advised me to use loc, I tested it and it worked with no problem. If you know of a better way then I could try that.

ExcessiveUseOfCobblestone commented:
You were able to get `isplasmaman` set to true?

ExcessiveUseOfCobblestone commented:
It almost looks like you can subtype box/survival for a """default""" box then use box/survival for universal items (which in this case would be the internal tank) and call inheritance.

willox commented:
@ExcessiveUseOfCobblestone backpack contents are spawned in a way so that `loc = the mob` and then get equipped into the backpack afterwards. https://github.com/tgstation/tgstation/blob/45458d74f1e33a1c0fc1be6fe6f53894542bd1d2/code/datums/outfit.dm#L218

ExcessiveUseOfCobblestone commented:
neat.

Now, regarding the copypasta... I'd really like this to be modular enough where you're not having to type the same thing for every subtype but that's prob too CI-ish to require for this PR.

pireamaineach commented:
Hey just saying it's probably not the best idea to give them a tank in their box that lasts ages longer than human emergency internals.
